src/components/carousel/index.js

```diff
@@ -0,0 +1,209 @@
+import React from "react"
+
+import classnames from "classnames"
+import withWindow from "../../utils/withWindow"
+import useCurrentWitdh from "../../utils/useCurrentWitdh"
+
+//
+// images: array of image. { url, title }
+//
+function Carousel({ images = [], height, onClickRoot }) {
+  const [width, setWidth] = React.useState(0)
+  const [isHoverMode, setIsHoverMode] = React.useState(false)
+  const [currentIndex, setCurrentIndex] = React.useState(0)
+
+  const windowWidth = useCurrentWitdh()
+
+  const nbImages = images.length
+
+  const rootRef = React.useRef(null)
+
+  //
+  // set isHoverMode to true and hide it after some time
+  //
+  let timeoutId = null
+  const activateHoverMode = () => {
+    setIsHoverMode(true)
+    clearTimeout(timeoutId)
+    timeoutId = setTimeout(() => {
+      setIsHoverMode(false)
+    }, 600)
+  }
+
+  //
+  // currentIndex utilities
+  //
+
+  const previous = () => {
+    if (currentIndex > 0) {
+      setCurrentIndex(currentIndex - 1)
+    } else {
+      setCurrentIndex(nbImages - 1)
+    }
+  }
+
+  const next = () => {
+    if (currentIndex + 1 === nbImages) {
+      setCurrentIndex(0)
+    } else {
+      setCurrentIndex(currentIndex + 1)
+    }
+  }
+
+  //
+  // window resize handling
+  //
+
+  // retrieve our carousel width
+  // we use the windowWidth to recalculate our element width when the window width changed
+  // (is resized).
+  React.useEffect(() => {
+    setWidth(rootRef.current?.offsetWidth || 0)
+  }, [rootRef.current, setWidth, windowWidth])
+
+  //
+  // keyboard shortcuts handling
+  //
+
+  const handleKeyDown = React.useCallback(
+    e => {
+      // arrow up/down button should select next/previous list element.
+      // we also use vim-keys (h,j,k,l)
+      if (
+        // up
+        e.keyCode === 38 ||
```

Comment:
Yes this is far better.
Just a question. What is the policy about browser support? Which browsers / minimum version should we handle?

Reply:
@loiclefloch we haven't talked about it but I think shooting for "modern browsers" roughly. Did you have any particular concerns with this PR?

Reply:
For example `indexOf` could be replaced by `includes` but it is not supported by IE. 
My main project being chrome-only I began to use `includes` before realizing it could be problematic on some browsers. And this is the case x)

So I think it would be nice that we define a policy for that :)

Reply:
Ah yeah, that's a good example. I wonder if Babel isn't already polyfilling stuff like this for us? I'm not sure..

Anyway, I'm not too worried about IE tbh.